src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -713,5 +724,32 @@ protected override bool TryParseRequest(ReadResult result, out bool endConnectio
         }
     }
 
+#pragma warning disable CS0618 // Type or member is obsolete
+    private void DetectHttp2Preface(ReadOnlySequence<byte> requestData, BadHttpRequestException ex)
+#pragma warning restore CS0618 // Type or member is obsolete
+    {
+        const int PrefaceLineLength = 18;
+
+        if (ConnectionFeatures.Get<ITlsHandshakeFeature>() == null)
```

Comment:
We don't need this check, the mistake is the same with or without TLS, even if non-TLS is more likely.

Reply:
Should I remove the same check from https://github.com/dotnet/aspnetcore/pull/39358?

The reason I have it is that not using TLS and then sending the wrong HTTP version is a user error.
Using TLS, negotiating an HTTP version, and then sending the wrong HTTP version seems more like a library user.

Reply:
I guess that makes sense, nevermind.